### DISCUSSION TITLE
Show dotfiles/hidden files by default

### DISCUSSION THREAD

**Discussion by m-zaremba** [nvim-tree/nvim-tree.lua#1287] [Q&A]
Is there a way to display hidden files by default and not by pressing 'I' (for some reason my '.env' file is treated as a hidden file, while .gitignore as a 'dotfile')?

**Comment by alex-courtis** [comment]
Hidden files (beginning with `.`) are shown by default: `filters.dotfiles = false`. `H` will toggle these.
See `:help nvim-tree.filters.dotfiles`

It sounds like your .env file is ignored by git - perhaps in your `~/.gitignore`.

**Reply by m-zaremba** [reply]
Sorry for not being precise enough. My situation is like this - in my React project's root directory I have two files - .gitignore and .env. Even though both are dotfiles, only .gitignore is visible by default (it can be hidden with 'H'). The .env file (and node_modules folder) is initially hidden and can be uncovered with 'I' ('H' has no effect).

As from the key bindings section:

- ```I``` will toggle visibility of hidden folders/files
- ```H``` will toggle visibility of dotfiles (files/folders starting with a ```.```)

I understand that there is a difference between hidden files/folders and dotfiles so I'm wondering if there is a possibility to make the former visible by default.

**Comment by alex-courtis** [comment]
> The .env file (and node_modules folder) is initially hidden and can be uncovered with 'I' ('H' has no effect).

`.env` is being considered ignored `I` by git and is not visible at startup.

Please run `git status --ignored` to verify this.

> I will toggle visibility of hidden folders/files

I see the problem. The README is incorrect. It will be updated #1291. See `:help nvim-tree-mappings` for a correct description of the actions.

**Comment by alex-courtis** [comment]
Unfortunately the above probably doesn't answer your question.

If `.env` is not being considered ignored by git we will need to investigate via replication:
- provide a git repo that can demonstrate the issue
- create an `nvt-min.lua` as per a [bug report](https://github.com/kyazdani42/nvim-tree.lua/issues/new?assignees=&labels=bug&template=bug_report.yml) that demonstrates
- write instructions on how to replicate the issue

**Reply by kyazdani42** [reply]
i believe the .env is just ignored by git because it's common in js/ts projects

**Reply by jettandres** [reply]
+1 for this. it's common for js/ts/rails projects to have a `.env` file at their root

**Reply by kyazdani42** [reply]
and still, there is a `:help nvim-tree.filters.exclude` so just play with that :smile:

**Comment by rvdende** [comment]
Ran into this and the fix is to set git = { ignore = false }

Here is my nvim-tree.lua file:

```lua
-- import nvim-tree plugin safely
local setup, nvimtree = pcall(require, "nvim-tree")
if not setup then
  return
end

-- recommended settings from nvim-tree documentation
vim.g.loaded_netrw = 1
vim.g.loaded_netrwPlugin = 1

-- change color for arrows in tree to light blue
vim.cmd([[ highlight NvimTreeIndentMarker guifg=#3FC5FF ]])

-- configure nvim-tree
nvimtree.setup({
  -- change folder arrow icons
  renderer = {
    icons = {
      glyphs = {
        folder = {
          arrow_closed = "+", -- arrow when folder is closed
          arrow_open = "-", -- arrow when folder is open
        },
      },
    },
  },
  -- disable window_picker for
  -- explorer to work well with
  -- window splits
  actions = {
    open_file = {
      window_picker = {
        enable = false,
      },
    },
  },
  git = {
    ignore = false,
  },
})
```

**Comment by zar3bski** [comment]
This is an old issue but it feels actual. I tried all the combination of 

```lua
return {
  {
    "nvim-tree/nvim-tree.lua",
    enabled = true,
    version = "*",
    lazy = false,
    dependencies = { "nvim-tree/nvim-web-devicons" },
    config = function()
      require("nvim-tree").setup({
        filters = {
          --- display dot files in file explorer, hopefully
          dotfiles = true,
          git_ignored = false,
        },
        git = {
          ignore = false,
        },
      })
    end,
  },
}
```
without any chance. I'm new to nvim so I do not grasp all the details but I know this conf is parsed (by crashing on purpose with random keys)

**Reply by alex-courtis** [reply]
Try `dotfiles = false`

`filters` are "positive" - when they are true, the filter is applied.

**Reply by zar3bski** [reply]
I precised '**I tried all the combinations of**'. So, no matter if `true` or `false` (I get you're point but does it moves the semantic ambiguity one step further: does a filter applied on a set of files means that they are shown or hidden?), I did not observe any change after I restart. With 

```lua
return {
  {
    "nvim-tree/nvim-tree.lua",
    enabled = true,
    version = "*",
    lazy = false,
    dependencies = { "nvim-tree/nvim-web-devicons" },
    config = function()
      require("nvim-tree").setup({
        filters = {
          --- display dot files in file explorer
          dotfiles = false,
          git_ignored = false,
        },
      })
    end,
  },
}
```
after I restart `nvim` none of the `.<files|folders>` appear by default. Pressing `h` `i` is not a big deal *per se* but since the functionalities seemed exposed in the configuration and the behavior did not seem right based on what I read on the documentation, I had to check if I was just dumb as f*. What am I missing? Do these parameters depend on other I did not identify?

**Reply by alex-courtis** [reply]
I've tested:

```sh
touch .dotf
touch no-dotf

mkdir .dotd
touch .dotd/.dotted
touch .dotd/no-dotted

mkdir nodotd
touch nodotd/.dotted
touch nodotd/no-dotted
```

With defaults all files and directories are shown, as expected. With your config, all are shown as expected.

There's some strange behaviour here that we need to resolve, maybe a bug.

Please raise a [Bug Report](https://github.com/nvim-tree/nvim-tree.lua/issues/new/choose) with all details provides so that we may replicate. Please use the clean room replication template `nvt-min.lua`, not a plugin manager, so that we can isolate the problem.

**Reply by zar3bski** [reply]
@alex-courtis : I finally understood what was wrong and I'm **deeply sorry for this false alarm**. The file explorer displayed in nvim was simply not `nvim-tree` but some file explorer activated by `Lazy` (haven't found who it is, nor how to turn it down yet, I wonder if this is the default nvim explorer and why `nvim-tree` does not appear by default) displayed over/instead of `nvim-tree`. Since it responded to the same keyboard shortcut, the noob I am wrongly assumed it was `nvim-tree`.

> I had to check if I was just dumb as f*

Looks like I am

**Reply by alex-courtis** [reply]
Nobody is dumb, it's a reasonable question.
